hexrd/imageseries/load/imagefiles.py

```diff
@@ -81,13 +81,28 @@ def __str__(self):
             self.dtype, self.shape, self.singleframes)
         return s
 
+    @property
+    def fname(self):
+        return self._fname
+
     def _load_yml(self):
         EMPTY = 'empty-frames'
         MAXTOTF = 'max-total-frames'
         MAXFILF = 'max-file-frames'
         DTYPE = 'dtype'
-        with open(self._fname, "r") as f:
-            d = yaml.safe_load(f)
+        #
+        # Check whether fname is a pathlib Path, a filename or YAML content.
+        # If it has multiple lines, we consider it to be YAML content,
+        # otherwise a file name.
+        #
+        is_str = isinstance(self.fname, str)
+        nlines = len(self.fname.splitlines()) if is_str else 1
+        if nlines > 1:
```

Comment:
Another alternative could be to check `os.path.exists(self.fname)`. If it's yaml, it wouldn't be an existing file path. But I think what is here works fine too!

Reply:
I thought of checking whether the path exists first but chose not to do it because of the case when the user passes a filename but might have a mistake in it. In that case, the file would not exist, and so it would be treated as YAML text. Then I'm not sure what the error would be, probably some key not found in the YAML, when it really should be a `FileNotFoundError`.  That is why I first try to determine if is YAML text, then if it looks like a file name, it will try to load and return the `FileNotFoundError` if it does not exist.

Reply:
That makes sense, Don

Reply:
Thanks also for the suggestion below with the pytest fixture for the data file in the examples repo. I didn't realize that was there and in the process learned about the pytest `conftest.py` file. That would be good to use, but at the moment, there are no image data files in the examples repo that would work because I am working on the 
`image-files` type of imageseries, and all the images in the examples are frame cache or HDF5.  

As I mentioned above, that gives me even more reason to set up some some examples of processing raw image data to make the frame caches.  Once I do that, then we could take advantage of it in the hexrd testing.

Reply:
Sounds great, Don. Thank you!